goldilocks/base.go

```diff
@@ -105,30 +113,38 @@ func New(api frontend.API) *Chip {
 		return chip
 	}
 
-	useBitDecomp := os.Getenv("USE_BIT_DECOMPOSITION_RANGE_CHECK")
-
 	c := &Chip{api: api}
 
-	// If USE_BIT_DECOMPOSITION_RANGE_CHECK is not set, then use the std.rangecheck New function
-	if useBitDecomp == "" {
-		c.usingCommitChecker = rangeCheckGadgetUsingCommitChecker(api)
-		c.usingSimpleChecker = false
-	} else {
-		c.usingCommitChecker = false
-		c.usingSimpleChecker = true
-	}
+	// Instantiate the range checker gadget
+	// Per Gnark's range checker gadget's New function, there are three possible range checkers:
+	// 1. The native range checker
+	// 2. The commit range checker
+	// 3. The bit decomposition range checker
+	//
+	// See https://github.com/Consensys/gnark/blob/3421eaa7d544286abf3de8c46282b8d4da6d5da0/std/rangecheck/rangecheck.go#L3
 
-	if c.usingCommitChecker || c.usingSimpleChecker {
-		c.simpleChecker = bitDecompChecker{api: api}
-	}
+	// This function will emulate gnark's range checker selection logic (within the gnarkRangeCheckSelector func).  However,
+	// if the USE_BIT_DECOMPOSITION_RANGE_CHECK env var is set, then it will explicitly use the bit decomposition range checker.
 
-	if c.usingCommitChecker {
-		api.Compiler().Defer(c.checkCollected)
+	rangeCheckerType := gnarkRangeCheckerSelector(api)
+	useBitDecomp := os.Getenv("USE_BIT_DECOMPOSITION_RANGE_CHECK")
+	if useBitDecomp == "true" {
+		rangeCheckerType = BIT_DECOMP_RANGE_CHECKER
```

Comment:
nit: maybe we can print out "Using bit decomp range checker" or something like that?